tests/test_txfunc.py

```diff
@@ -731,6 +734,66 @@ def test_monotone_spline():
     """
     # Simulate some data
     np.random.seed(10)
+
+    # another test case for pygam method:
+
+    # N = 100
+    # xlo = 0.001
+    # xhi = 2 * np.pi
+    # x1 = np.arange(xlo, xhi, step=(xhi - xlo) / N)
+    # x2 = np.arange(xlo, xhi, step=(xhi - xlo) / N)
+    # x = np.array([x1, x2]).T
+    # X1, X2 = np.meshgrid(x1, x2)
+    # X1, X2 = X1.flatten(), X2.flatten()
+    # X = np.zeros((X1.shape[0], 2))
+    # X[:,0] = X1
+    # X[:,1] = X2
+    # y0 = np.exp(np.sin(X1)) * np.exp(np.cos(X2))
+    # y0 = np.cos(X1)/(1.01 + np.sin(X2))
+    # y0 = 1/(1 + np.exp(-(X1+X2)))
+    # y = y0 + np.random.random(y0.shape) * 0.2
+    # weights = np.ones(10000)
+    # (
+    # mono_interp,
+    # y_cstr,
+    # wsse_cstr,
+    # y_uncstr,
+    # wsse_uncstr,
+    # ) = txfunc.monotone_spline(
+    # X, y, weights, lam=100, incl_uncstr=True, show_plot=True, method='pygam'
+    # )
+
+    with open("test_io_data/micro_data_dict_for_tests.pkl", "rb") as f:
```

Comment:
Let's use `os.path.join(CUR_PATH, "test_io_data", "micro_data_dict_for_tests.pkl")` for the files name so we can open it from other directories besides `tests` (e.g., this test fails to open the pickle file when I run `pytest` from `OG-Core/`

Also, should we make use of the `utils.safe_read_pickle` as we do elsewhere?

Reply:
@prrathi I see you removed your local path.  That's helpful.  But also note the suggested changes to specify paths from the `CUR_PATH`.

Reply:
@jdebacker @rickecon added the change. Also from testing locally I think this needs numpy version < 1.24, because the pygam package has code that doesn't work with numpy >= 1.24, specifically they call numpy.int which seems to be deprecated.